packages/type-storage/src/utils/createFunction.ts

```diff
@@ -51,6 +51,20 @@ export default function createFunction (section: Text | string, method: Text | s
     );
   };
 
+  if (meta.type.isMap && meta.type.asMap.isLinked) {
+    // TODO: there needs some better way to do this
+    const keyHash = new U8a(xxhashAsU8a(`head of ${stringKey}`, 128));
```

Comment:
this code is bit messy
`new U8a` is needed here to indicate the data does not have length prefix
I need this `keyFn` function to specify the output format, which is the key type

Reply:
You can just use `xxhashAsHex` without the `new U8a()` wrapping.